client/allocdir/task_dir.go

```diff
@@ -184,12 +183,13 @@ func (t *TaskDir) embedDirs(entries map[string]string) error {
 		}
 
 		// Enumerate the files in source.
-		dirEntries, err := ioutil.ReadDir(source)
+		dirEntries, err := os.ReadDir(source)
 		if err != nil {
 			return fmt.Errorf("Couldn't read directory %v: %v", source, err)
 		}
 
-		for _, entry := range dirEntries {
+		for _, fileEntry := range dirEntries {
+			entry, _ := fileEntry.Info()
```

Comment:
I think we need to not ignore this error - it would have previously been reported by `ioutil.ReadDir`

The deprecation notice suggests the same, 
https://github.com/golang/go/blob/master/src/io/ioutil/ioutil.go#L59-L68